src/igvfd/audit/measurement_set.py

```diff
@@ -266,6 +266,77 @@ def audit_missing_institutional_certification(value, system):
             yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level=audit_message.get('audit_level', ''))
 
 
+@audit_checker('MeasurementSet', frame='object')
+def audit_inconsistent_controlled_access(value, system):
+    '''
+    [
+        {
+            "audit_description": "The files in a measurement set should be submitted with the same access restrictions as indicated by the institutional certificate covering the samples linked to the same measurement set.",
+            "audit_category": "inconsistent controlled access",
+            "audit_level": "ERROR"
+        },
+        {
+            "audit_description": "All files in a measurement set should be submitted with the same access restrictions as each other.",
+            "audit_category": "inconsistent controlled access",
+            "audit_level": "ERROR"
+        }
+    ]
+    '''
+    audit_message_file_sample = get_audit_message(audit_inconsistent_controlled_access, index=0)
+    audit_message_mixed_files = get_audit_message(audit_inconsistent_controlled_access, index=1)
+
+    files_by_access = {}
+    files = value.get('files', [])
+    for file in files:
+        file_object = system.get('request').embed(file, '@@object')
+        if file_object.get('controlled_access', '') not in files_by_access:
+            files_by_access[file_object.get('controlled_access', '')] = [file_object.get('@id')]
+        else:
+            files_by_access[file_object.get('controlled_access', '')].append(file_object.get('@id'))
+
+    if True in files_by_access and False in files_by_access:
+        controlled_access_of_files_phrase = 'controlled and uncontrolled access'
+        controlled_files_link = ', '.join([audit_link(path_to_text(file), file) for file in files_by_access[True]])
+        uncontrolled_files_link = ', '.join([audit_link(path_to_text(file), file) for file in files_by_access[False]])
+        detail = (
+            f'Measurement set {audit_link(path_to_text(value["@id"]), value["@id"])} contains '
+            f'files {controlled_files_link} submitted as controlled access and files {uncontrolled_files_link} '
+            f'submitted as uncontrolled access.'
+        )
+        yield AuditFailure(
+            audit_message_mixed_files.get('audit_category', ''),
+            f'{detail} {audit_message_mixed_files.get("audit_description", "")}',
+            level=audit_message_mixed_files.get('audit_level', '')
+        )
+    elif True in files_by_access and False not in files_by_access:
+        controlled_access_of_files_phrase = 'controlled access'
+    elif True not in files_by_access and False in files_by_access:
+        controlled_access_of_files_phrase = 'uncontrolled access'
+
```

Comment:
I was looking at if controlled_access is a required boolean field for all files. In index file and model files they are optional, config/matrix/signal files they don't have that property. So theoretically could be a case where all files don't have `controlled_access` and the key in `files_by_access` is only an empty string -> `controlled_access_of_files_phrase` will not be assigned before it might be called below. This seems to be an invalid case for meaSet, but I think you could skip files without `controlled_access` in files_by_access to be safe.

Reply:
Also I feel maybe we should make controlled_access as a required field for index file, or be calculated from its derived_from file, which could be alignment file.

Reply:
I wonder if that is possible to change since I am not sure we could upgrade Index File. I guess this could be a future ticket.